app/controllers/collections_controller.rb

```diff
@@ -131,12 +127,74 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def create_collection_core(name, metadata, owner)
+    name = sanitize_collection_name(name)
+    metadata = update_jsonld_collection_id(metadata, name)
+    uri = metadata['@id']
+    if Collection.find_by_uri(uri).present?  # ingest skips collections with non-unique uri
+      raise ResponseError.new(400), "A collection with the name '#{name}' already exists"
+    else
+      corpus_dir = create_metadata_and_manifest(name, convert_json_metadata_to_rdf(metadata))
+      # Create the collection without doing a full ingest since it won't contain any item metadata
+      collection = check_and_create_collection(name, corpus_dir)
+      collection.owner = owner
+      collection.save
+      "New collection '#{name}' (#{uri}) created"
+    end
+  end
+
+  def web_create_collection
+    authorize! :web_create_collection, Collection
+    # Provide public licences and the user licences
+    t = Licence.arel_table
+    @licences = Licence.where(t[:private].eq(false).or(t[:owner_id].eq(current_user.id)))
+    # Handle a submitted collection creation form
+    if request.post?
+      required_fields = {:collection_name => 'collection name', :collection_title => 'collection title'}
+      protected_field_keys = ['@id', '@type', '@context',
+                             'dc:identifier', 'dcterms:identifier', MetadataHelper::IDENTIFIER.to_s,
+                             'dc:title', 'dcterms:title', MetadataHelper::TITLE.to_s]
+      begin
+        required_fields.each do |key, value|
+          raise ResponseError.new(400), "Required field '#{value}' is missing" if params[key].blank?
+        end
+        name = sanitize_collection_name(params[:collection_name])
+        title = params[:collection_title]
+        json_ld = {'@context' => JsonLdHelper::default_context, '@type' => 'dcmitype:Collection', MetadataHelper::TITLE.to_s => title}
+        # Validate and sanitise any additional metadata fields
+        additional_metadata = []
+        if params.has_key?(:additional_key) && params.has_key?(:additional_value)
+          additional_metadata = params[:additional_key].zip(params[:additional_value])
+          additional_metadata.delete(['','']) # delete any pairs with empty key and value
+          additional_metadata.each do |key, value|
+            unless key.blank? && value.blank?
+              raise ResponseError.new(400), 'An additional metadata field is missing a key' if key.blank?
+              raise ResponseError.new(400), "Additional metadata field '#{key}' is missing a value" if value.blank?
+              json_ld[key.strip] = value.strip unless protected_field_keys.include?(key.strip)
+            end
+          end
+        end
+        # Ingest new collection and assign a licence
+        json_ld = JSON.parse(json_ld.to_json) # convert symbols in context to strings as expected in JSON-LD parser
```

Comment:
Using strings as keys instead of symbols would mean that conversion wouldn't be needed. I just kept it as it was to minimise the changes I introduced.